ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -264,6 +264,13 @@ let check_protection_policy ~vmr ~op ~ref_str =
                                  [ref_str; Ref.string_of vmr.Db_actions.vM_protection_policy])
   | _ -> None
 
+(* VM cannot be converted into a template while it is assigned to a snapshot schedule. *)
+let check_snapshot_schedule ~vmr ~op ~ref_str =
```

Comment:
By make `op` the last parameter, this could be written as 

```
let check_snapshot_schedule ~vmr ~ref_str = function
|  `make_into_template ->
| _ -> None
```